src/esp/sim/Simulator.h

```diff
@@ -848,7 +848,63 @@ class Simulator {
  protected:
   Simulator(){};
 
-  //! sample a random valid AgentState in passed agentState
+  /**
+   * @brief Creates a pathfinder and loads the navmesh from the specified
+   * location if it exists, otherwise returns a warning.
+   * @param navmeshFilename The filename of the navmesh to load when pathfinder
+   * is created
+   * @return Whether successful or not
+   */
+  bool createPathfinder(const std::string& navmeshFilename);
```

Comment:
Wow, why is the necessary over the existing pathfinder API? shouldn't this be a method on pathfinder? Why should it be its own method. (There are already a lot of duplicate / unused methods on Simulator.h, so we should reuse / refactor when possible.)

Reply:
This is to support the eventual addition of multiple navmeshes, each of which will require their own pathfinder.

Reply:
Still should remove the amount stat calls to the file system. Doesn't pathfinder already do a file exists check?

Reply:
I agree that fewer OS hits would be a great thing.  What you see was the existing mechanism for loading navmesh in reconfigure, which is why I have not modified it.  

The loading function does check for file existence, along with multiple other failure conditions, and returns only a boolean success value.  By having this test, we can know for certain whether failure is caused by non-existing file, or some other condition, without having to refactor PathFinder::loadNavMesh, which I felt is a bit beyond the scope of this PR.